src/util/observatory.ts

```diff
@@ -0,0 +1,63 @@
+import {BoaviztaCpuImpactModel} from '../lib';
+
+/**
+ * Pipeline for computing impacts based on observation.
+ */
+export class Observatory {
+  private observations: any;
+
+  /**
+   * Init observations object.
+   */
+  constructor(observations: any) {
+    this.observations = observations;
+  }
+
+  /**
+   * Apply `boavizta` model computation to observations.
+   */
+  private async boaviztaHandler(params: any) {
+    const modelInstance = await new BoaviztaCpuImpactModel().configure(
+      'test',
+      params
+    );
+
+    const preparedObservations = this.observations.map((observation: any) => ({
+      cpu: observation.cpu,
+      duration: `${observation.duration}s`,
+      datetime: observation.timestamp,
+    }));
+
+    const calculatedImpacts = await modelInstance.calculate(
+      preparedObservations
+    );
+
+    const result = this.observations.map((observation: any, index: number) => ({
+      ...observation,
+      ...calculatedImpacts[index],
+    }));
+
+    this.observations = result;
```

Comment:
Store the ouput in a thing called impacts, yes it will cotain the same data as the observations but any impacts added in.

impacts:
...
obsevations:
...